Lab10/closedunidirectionallinkedlist.py

```diff
@@ -0,0 +1,198 @@
+from device import Device
+
+
+class Node:
+    def __init__(self, data: Device = None) -> None:
+        self.data: Device = data
+        self.next: Node = None
+
+
+class ClosedUnidirectionalLinkedList:
+    def __init__(self) -> None:
+        self.head: Node = Node(None)
+        self.tail: Node = Node(None)
+        self.head.next = self.tail
+        self.tail.next = self.head
+
+    # add -> add some element type Device to list
+    def add(self, data: Device) -> None:
+        new_node: Node = Node(data)
+        if self.head.data is None:
+            self.head = new_node
+            self.tail = new_node
+            new_node.next = self.head
+        else:
+            self.tail.next = new_node
+            self.tail = new_node
+            self.tail.next = self.head
+
+    # print_list -> print all elements of list
+    def print_list(self) -> None:
+        current: Node = self.head
+        if self.head.data is None:
+            print("List is empty!")
```

Comment:
It's not necessary for this task, although this print could be a [custom exception](https://www.programiz.com/python-programming/user-defined-exception) :)